app/util/route.js

```diff
@@ -0,0 +1,89 @@
+import {equals} from 'ramda'
+import PreactRouter from 'preact-router'
+import {map, reduce} from 'wasmuth'
+import compose from '/util/compose'
+import {set, dispatch, getState} from '/store'
+import {routes} from '/routes'
+
+const {cloneElement} = Preact
+
+/**
+ * Use instaed of preact-router Router component.
+ * All children should be Route components (defined below).
+ * - Add key prop to Route children so that when route changes,
+ *   the current route component will unmount and the new one will mount.
+ * - Add path prop to Route children based on their name props.
+ */
+export const Router = ({children, routes, ...props}) =>
+  <PreactRouter {...props}>
+    {map((route, i) => cloneElement(route, {
+      key: i,
+      path: routes[route.attributes.name].path
+    }), children)}
+  </PreactRouter>
+
+/**
+ * Add preact-router props into the atom state
```

Comment:
Might be worth documenting a bit more -- is my understanding correct:

Router wraps PreactRouter and clones all children to add the react-router path value as a prop on those children.

Also, would it be worth enforcing all children of Router are Route's? Or, only cloning if they are an instance of Route?

Reply:
"Router wraps PreactRouter and clones all children to add the *routes* path *(that corresponds tot he child's name prop)* value as a prop on those children."

Reply:
Updating so that all children must be Route components